src/Journal.h

```diff
@@ -86,7 +86,7 @@ namespace Journal
    //\brief thrown when playback of a journal doesn't match the recording
    class SyncException : public AudacityException {
    public:
-      SyncException();
+      explicit SyncException(const wxString& message);
```

Comment:
I see no nullary ctor any more, so this forced rewrite of all uses.  That's good.  I would just like a comment explaning that the message is for the side logging purposes only and is not written to the output journal.